src/modules/auth/domain/auth.service.ts

```diff
@@ -0,0 +1,79 @@
+import { ClassProvider, Inject, Injectable, HttpException,  HttpStatus} from "@nestjs/common";
+import {
+	ServiceLoginInput,
+	ServiceCheckTokenInput,
+	ServiceAuth,
+	AuthServiceDefinition,
+	AuthServiceName,
+} from "src/modules/auth/domain/auth.service.types";
+import { UserServiceDefinition, UserServiceName, ServiceUser } from "src/modules/user/domain/user.service.types";
+import {
+	RepoAuth,
+	AuthRepositoryDefinition,
+	AuthRepositoryName,
+} from "src/modules/auth/repository/auth.repository.types";
+import CryptoJS = require("crypto-js"); // hash
+import crypto = require("crypto"); // token
+
+const VALID_TOKEN_TIME = 2628n * 10000n; // Amount of seconds in a mounth
+export const SALT = "3b7f9e2c1d4a5e6b7c8f0a1d9e3b2c4d";
```

Comment:
I don't believe this is being done properly. Salt's aren't statically defined like this. 

Also, the VALID_TOKEN_TIME wouldn't be defined here. It would be defined in an environment variable